pkg/sql/schema_adapter.go

```diff
@@ -25,6 +25,9 @@ type SchemaAdapter interface {
 	// SchemaInitializingQueries returns SQL queries which will make sure (CREATE IF NOT EXISTS)
 	// that the appropriate tables exist to write messages to the given topic.
 	SchemaInitializingQueries(topic string) []string
+	// ShouldIgnore indicates that this error should not be considered as hard error and we should simply retry
+	// this is useful for deadlocks and other serialization errors.
+	ShouldIgnore(err error) bool
```

Comment:
As `SchemaAdapter` is exported, extending this interface is a breaking change and would need a major version bump.

Perhaps we could add another interface with this method, so it doesn't break the current contract? It could be added to the config, next to `SchemaAdapter`, and filled with default if not provided.